### PR TITLE
Full salary breakdown

### PR DESCRIPTION
**Pull request by abhishekkhanduri**
#2026

### Description

- Complete breakdown of Salary
- Added fields in salary setting page to store health insurance amount and  employer Esi limit
- We can also add decimal values in setting page
- Health insurance amount will only be applicable incase where there will be no employee Esi or no employer Esi

### Checklist:

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue [hunk 1]
```diff
@@ -46,6 +48,122 @@
...
 			</div>
 		</div>
+		<h2 class="text-underline">Salary Deduction</h2>
+		<br>
+		<div class="row mb-5">
+			<div class="col-md-4">
+				<div class="text-secondary mb-1">Employee ESI</div>
+				<div class="fz-30">
+					<i class="fa fa-rupee"></i>
+					<span>NA</span>
```
Comment (rathorevaibhav) [inline Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue:58]: please fix at all places
```suggestion
					<span>N/A</span>
```

File: Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue [hunk 2]
```diff
@@ -46,6 +48,122 @@
 				</div>
 			</div>
 		</div>
+		<h2 class="text-underline">Salary Deduction</h2>
+		<br>
+		<div class="row mb-5">
+			<div class="col-md-4">
+				<div class="text-secondary mb-1">Employee ESI</div>
+				<div class="fz-30">
+					<i class="fa fa-rupee"></i>
```
Comment (rathorevaibhav) [inline Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue:57]: let's remove rupee sign where we're hardcoding NA

File: Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue [hunk 3]
```diff
@@ -46,6 +48,122 @@
...
+				<div class="text-secondary mb-1">Health Insurance</div>
+				<div class="fz-30">
+					<i class="fa fa-rupee"></i>
+					<span>{{ this.formatCurrency(healthInsurance) }}</span>
+				</div>
+			</div>
+		</div>
+		<div class="row mb-5">
+			<div class="col-md-4">
+				<div class="text-secondary mb-1">CTC Aggreed</div>
```
Comment (rathorevaibhav) [inline Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue:160]: let's name it `CTC Aggregated`

File: Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue [hunk 4]
```diff
@@ -46,6 +48,122 @@
...
+				<div class="fz-30">
+					<i class="fa fa-rupee"></i>
+					<span>{{ this.formatCurrency(ctcAggreed) }}</span>
```
Comment (rathorevaibhav) [inline Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue:163]:
```suggestion
					<span>{{ this.formatCurrency(ctcAggregated) }}</span>
```

File: Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue [hunk 5]
```diff
@@ -84,8 +202,60 @@ export default {
...
+			return Math.ceil(multiplier * percentage / 100);
+		},
+		totalDeduction() {
+			return this.employeeEpf + this.foodAllowance;
+		},
+		netPay() {
+			return this.totalSalary - this.totalDeduction;		
+		},
+		employerEsi() {
+			if(this.grossSalary< 21000) {
```
Comment (rathorevaibhav) [inline Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue:220]:
```suggestion
			if (this.grossSalary < 21000) {
```

File: Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue [hunk 6]
```diff
@@ -84,8 +202,60 @@ export default {
...
+			let percentage = this.salaryConfigs.edli_charges.percentage_rate;
+			return Math.min(Math.ceil(multiplier * percentage / 100) , Math.ceil(this.salaryConfigs.edli_charges_limit.fixed_amount * percentage / 100));
+		},
+		ctc() {
+			return Math.ceil(parseInt(this.grossSalary) + parseInt(this.employerEsi) + parseInt(this.employeeEpf) + parseInt(this.edliCharges) + parseInt(this.administrationCharges));
+		},
+		ctcAnnual() {
+			return this.ctc * 12;		
+		},
+		healthInsurance() {
```
Comment (rathorevaibhav) [inline Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue:248]: healthinsurnace is also zero if employerESI or employeeESI is non-zero/non-null

Comment (abhishekkhanduri) [reply Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue:262]: ok

File: Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue (hunk 5 above)
Comment (abhishekkhanduri) [reply Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue]: done

File: Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue (hunk 4 above)
Comment (abhishekkhanduri) [reply Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue]: done

File: Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue (hunk 2 above)
Comment (abhishekkhanduri) [reply Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue:57]: ok

File: Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue (hunk 3 above)
Comment (abhishekkhanduri) [reply Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue]: ok

File: Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue [hunk 7]
```diff
@@ -84,8 +201,78 @@ export default {
...
+			if (this.grossSalary === "") {
+				return 0;
+			}
+			if(this.employerEsi === null || this.emplpyeeEsi === null)
+			{
+				return parseInt(this.salaryConfigs.health_insurance.fixed_amount);
+			}
+			return 0;
```
Comment (rathorevaibhav) [inline Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue:270]: I think we can combine
```suggestion
			if (this.grossSalary === "" || this.employerEsi === null || this.emplpyeeEsi === null) {
				return 0;
			}
			return parseInt(this.salaryConfigs.health_insurance.fixed_amount);
```

File: Modules/Salary/Resources/views/settings/index.blade.php [hunk 8]
```diff
@@ -29,83 +29,91 @@ class="form-control w-10" placeholder="%"
...
-                                            value="{{ optional($salaryConfig->get('hra', null))->percentage_rate }}">
+                                            value="{{ optional($salaryConfig->get('hra', null))->percentage_rate }}" step="0.01">
                                         <div class="input-group-append">
                                             <span class="input-group-text" id="basic-addon2">% of basic salary</span>
                                         </div>
                                     </div>
                                 </div>
                                 <div class="form-group col-md-6">
-                                    <label>Employee ESI</label>
+                                    <label>Employer ESI</label>
```
Comment (rathorevaibhav) [inline Modules/Salary/Resources/views/settings/index.blade.php:47]: Employer ESI and Employee ESI are both based on Gross salary. Confirm from the sheet. Which means they should be in Gross salary section

Comment (abhishekkhanduri) [reply Modules/Salary/Resources/views/settings/index.blade.php]: Yes, I have checked, Employer ESI and Employee ESI are both based on Gross salary so made changes according to that

File: Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue (hunk 7 above)
Comment (abhishekkhanduri) [reply Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue]: ok

File: Modules/Salary/Resources/views/settings/index.blade.php [hunk 9]
```diff
@@ -114,6 +94,42 @@ class="form-control w-10" placeholder="%"
                             </div>
                         </div>
                     </div>
+                    <div class="row mb-6">
+                        <div class="col-12">
+                            <div class="form-row">
+                                <div class="form-group col-md-6">
+                                    <h4 class="text-underline">Deduction based on gross salary</h4>
```
Comment (rathorevaibhav) [inline Modules/Salary/Resources/views/settings/index.blade.php:101]: can this be put under Gross salary section. So essentially here's the section breakdown:
1. Based on gross salary
   1. Fields
   2. Deductions based on gross salary (with small heading)
3. Based on basic salary
   1. Field
   2. Deductions based on gross salary (with small heading)
4. Fixed amounts

**Review by rathorevaibhav — APPROVED** [review]

**Pull request merged**